records/generator.go

```diff
@@ -21,6 +23,8 @@ import (
 // Will likely become map[string][]discoveryinfo
 type rrs map[string][]string
 
+var DefaultDomainPatterns = []string{"{{.TaskName}}"}
```

Comment:
I'd be tempted to make it package-private and then write a public func that returns a clone. I'd still use the package-private reference within the package for ease of use and readability. And maybe a comment for the var that mentions that it's intended to be read-only.